controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -306,13 +308,24 @@ func (r *DataScienceClusterReconciler) reportError(err error, instance *dsc.Data
 	return instance
 }
 
+var configMapPredicates = predicate.Funcs{
+	UpdateFunc: func(e event.UpdateEvent) bool {
+		fmt.Printf("Got update event for resource %v: %v", e.ObjectNew.GetObjectKind(), e.ObjectNew.GetName())
+		// Do not reconcile on prometheus configmap update, since it is handled by DSCI
+		if e.ObjectNew.GetName() == "prometheus" && e.ObjectNew.GetNamespace() == "redhat-ods-monitoring" {
+			return false
+		}
+		return true
+	},
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *DataScienceClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&dsc.DataScienceCluster{}).
 		Owns(&corev1.Namespace{}).
 		Owns(&corev1.Secret{}).
-		Owns(&corev1.ConfigMap{}).
+		Owns(&corev1.ConfigMap{}, builder.WithPredicates(configMapPredicates)).
```

Comment:
should the predicate be added to the ConfigMap Watches() call below as well?

Reply:
hum, maybe it is better to have this builder.WithPredicates() in the Watches() than Owns().

Reply:
We require it in `Owns()` since the configmap has DSC as ownerref

I also added it to watches, incase there are other configmaps with updates

Reply:
got it, thanks!